src/platform/linux/misc.cpp

```diff
@@ -100,22 +100,43 @@ namespace platf {
 
   fs::path
   appdata() {
+    bool found = false;
+    bool migrate_config = true;
     const char *dir;
+    const char *homedir;
+    fs::path config_path;
+
+    // Get the home directory
+    if ((homedir = getenv("HOME")) == nullptr) {
```

Comment:
While you're here, we should also check `&& strlen(homedir) > 0` and similar for all the `getenv()` stuff below.

The XDG specification says "not set or empty" and we're only checking the former right now.